crypto/ec/curve448/arch_64/arch_intrinsics.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017-2018 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2016 Cryptography Research, Inc.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ * Originally written by Mike Hamburg
+ */
+
+#ifndef OSSL_CRYPTO_EC_CURVE448_ARCH_64_INTRINSICS_H
+# define OSSL_CRYPTO_EC_CURVE448_ARCH_64_INTRINSICS_H
+
+#include "internal/constant_time.h"
+
+# define ARCH_WORD_BITS 64
+
+#define word_is_zero(a)     constant_time_is_zero_64(a)
+
+static ossl_inline __uint128_t widemul(uint64_t a, uint64_t b)
```

Comment:
For other fixed width int types we have standardised on `uintX_t` as the type name - which is mostly covered by systems that have inttypes.h. For systems that have the type, but it's called something else then we've provided suitable typedefs in "include/openssl/e_os2.h". I wonder if we should do the same here?

Reply:
For a similar use case of 128-bit integers for NIST prime elliptic curves there is a configuration variable `ec_nistp_64_gcc_128`.  Would something similar (`ec_x448_64_gcc_128`) make sense here?